stdlib/universe/moving_average_test.go

```diff
@@ -0,0 +1,521 @@
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+					{Label: "t", Type: flux.TString},
+				},
+				Data: [][]interface{}{
+					{execute.Time(3 * time.Second), 11.0 / 3, "c"},
+					{execute.Time(4 * time.Second), 6.0, "d"},
+					{execute.Time(5 * time.Second), 22.0 / 3, "e"},
+					{execute.Time(6 * time.Second), 28.0 / 3, "f"},
+					{execute.Time(7 * time.Second), 34.0 / 3, "g"},
+					{execute.Time(8 * time.Second), 38.0 / 3, "h"},
+					{execute.Time(9 * time.Second), 32.0 / 3, "i"},
+					{execute.Time(10 * time.Second), 8.0, "j"},
+				},
+			}},
+		},
+		{
+			name: "nulls in time column",
+			spec: &universe.MovingAverageProcedureSpec{
+				Columns: []string{"x", "y"},
+				N:       2,
+			},
+			data: []flux.Table{&executetest.Table{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "x", Type: flux.TFloat},
+					{Label: "y", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{nil, 2.0, 3.0},
+					{execute.Time(2), nil, 10.0},
+					{nil, 8.0, 20.0},
+					{execute.Time(4), 8.0, 20.0},
+					{nil, 8.0, 20.0},
+					{execute.Time(6), 10.0, 25.0},
+					{nil, 8.0, 20.0},
+				},
+			}},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "x", Type: flux.TFloat},
+					{Label: "y", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(2), 2.0, 13.0 / 2},
+					{nil, 8.0, 15.0},
+					{execute.Time(4), 8.0, 20.0},
+					{nil, 8.0, 20.0},
+					{execute.Time(6), 9.0, 22.5},
+					{nil, 9.0, 22.5},
+				},
+			}},
+		},
+		{
+			name: "mean average of non-numerical column",
+			spec: &universe.MovingAverageProcedureSpec{
+				Columns: []string{"x", "y"},
+				N:       2,
+			},
+			data: []flux.Table{&executetest.Table{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "x", Type: flux.TFloat},
+					{Label: "y", Type: flux.TString},
+				},
+			}},
+			wantErr: fmt.Errorf("cannot take moving average of column y (type string)"),
+		},
+		{
+			name: "int nulls",
+			spec: &universe.MovingAverageProcedureSpec{
+				Columns: []string{"x", "y", "z"},
+				N:       2,
+			},
+			data: []flux.Table{&executetest.Table{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "x", Type: flux.TInt},
+					{Label: "y", Type: flux.TInt},
+					{Label: "z", Type: flux.TInt},
+				},
+				Data: [][]interface{}{
+					{execute.Time(1), nil, int64(1), int64(2)},
+					{execute.Time(2), nil, int64(2), nil},
+					{execute.Time(3), int64(4), nil, int64(4)},
+					{execute.Time(4), int64(2), nil, int64(4)},
+				},
+			}},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "x", Type: flux.TFloat},
+					{Label: "y", Type: flux.TFloat},
+					{Label: "z", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(2), 0.0, 1.5, 2.0},
```

Comment:
The first `0.0` is still surprising. I would expect that the avg of `nil nil` is `nil`.